frameworks/cassandra/tests/test_tls.py

```diff
@@ -0,0 +1,256 @@
+def dcos_ca_bundle():
+    """
+    Retrieve DC/OS CA bundle and returns the content.
+    """
+    url = shakedown.dcos_url_path('ca/dcos-ca.crt')
+    resp = dcos.http.request('get', url)
+    cert = resp.content.decode('ascii')
+    assert cert is not None
+    return cert
+
+
+@pytest.fixture(scope='module')
+def service_account():
+    """
+    Creates service account with `hello-world` name and yields the name.
+    """
+    name = PACKAGE_NAME
+    sdk_security.create_service_account(
+        service_account_name=name, service_account_secret=name)
+     # TODO(mh): Fine grained permissions needs to be addressed in DCOS-16475
+    sdk_cmd.run_cli(
+        "security org groups add_user superusers {name}".format(name=name))
+    yield name
+    sdk_security.delete_service_account(
+        service_account_name=name, service_account_secret=name)
+
+
+@pytest.fixture(scope='module')
+def cassandra_service_tls(service_account):
+    sdk_install.install(
+        PACKAGE_NAME,
+        DEFAULT_TASK_COUNT,
+        service_name=service_account,
+        additional_options={
+            "service": {
+                "service_account_secret": service_account,
+                "service_account": service_account,
+                "tls": True,
+                "tls_allow_plaintext": False,
+            }
+        }
+    )
+
+    sdk_plan.wait_for_completed_deployment(PACKAGE_NAME)
+
+    # Wait for service health check to pass
+    shakedown.service_healthy(PACKAGE_NAME)
+
+    yield service_account
+
+    sdk_install.uninstall(PACKAGE_NAME)
+
+
+def get_cqlsh_tls_rc_config(
+        certfile='/mnt/mesos/sandbox/ca-bundle.crt',
+        hostname=DEFAULT_NODE_ADDRESS,
+        port=DEFAULT_NODE_PORT
+    ):
+    """
+    Returns a content of `cqlshrc` configuration file with provided hostname,
+    port and certfile location. The configuration can be used for connecting
+    to cassandra over a TLS connection.
+    """
+    return textwrap.dedent("""
+        [cql]
+        ; Substitute for the version of Cassandra you are connecting to.
+        version = 3.4.0
+
+        [connection]
+        hostname = {hostname}
+        port = {port}
+        factory = cqlshlib.ssl.ssl_transport_factory
+
+        [ssl]
+        certfile = {certfile}
+        ; Note: If validate = true then the certificate name must match the machine's hostname
+        validate = true
+        ; If using client authentication (require_client_auth = true in cassandra.yaml) you'll also need to point to your uesrkey and usercert.
+        ; SSL client authentication is only supported via cqlsh on C* 2.1 and greater.
+        ; This is disabled by default on all Instaclustr-managed clusters.
+        ; userkey = /path/to/userkey.pem
+        ; usercert = /path/to/usercert.pem
+        """.format(
+            hostname=hostname,
+            port=port,
+            certfile=certfile
+        ))
+
+
+def get_write_data_job(dcos_ca_bundle: str):
+    query = ' '.join([
```

Comment:
Would it make sense to also use `textwrap.dedent` here too?

Reply:
These are separate lines so there is no real indentation in the string.

Reply:
Looking at it in an editor instead of githubs diff view might make it a little clearer. One could place the closing `])` on a new line to make the lines even easier to distinguish.

Reply:
Moved closing `])` to the separate line.